packages/App/src/Registaration/LogIn.tsx

```diff
@@ -0,0 +1,215 @@
+  password: yup.string().required(),
+});
+
+const LogIn: FC = ({navigation}: any) => {
+  const HTTP = useContext(Context);
+
+  const [visiblePass, setVisiblePass] = useState<boolean>(true);
+
+  const logIn = async (emailReg: string, passwordReg: string) => {
+    const response = await HTTP.auth(
+      'http://localhost:8000/auth/login',
+      emailReg,
+      passwordReg,
+    );
+    console.log(response);
+  };
+
+  const signUpPage = () => {
+    navigation.navigate('SignIn');
+  };
+
+  return (
+    <ImageBackground style={styles.backGround} source={BG}>
+      <Text style={styles.title}>Log In</Text>
+
+      <Formik
+        initialValues={{email: '', password: ''}}
+        validationSchema={logInSchema}
+        onSubmit={values => {
+          logIn(values.email, values.password);
+        }}>
+        {({
+          handleChange,
+          handleBlur,
+          handleSubmit,
+          values,
+          errors,
+          touched,
+          isValid,
+          dirty,
+        }) => (
+          <>
+            <View style={styles.inputArea}>
+              <Text style={styles.label}>Email</Text>
+              <TextInput
+                style={styles.input}
+                onChangeText={handleChange('email')}
+                value={values.email}
+                textContentType="emailAddress"
+                keyboardType="email-address"
+                onBlur={handleBlur('email')}
+              />
+              {touched.email && errors.email && (
+                <Text style={styles.error}>{errors.email}</Text>
+              )}
+            </View>
+
+            <View style={styles.inputArea}>
+              <Text style={styles.label}>Password</Text>
+
+              <TextInput
+                style={styles.input}
+                onChangeText={handleChange('password')}
+                value={values.password}
+                textContentType="password"
+                secureTextEntry={visiblePass}
+                onBlur={handleBlur('password')}
+              />
+              <TouchableOpacity
+                onPress={() => setVisiblePass((prev: boolean) => !prev)}>
+                <Image
+                  source={visiblePass ? passwordUnvisible : passwordVisible}
+                  style={styles.passwordIcon}
+                />
+              </TouchableOpacity>
+              {touched.password && errors.password && (
+                <Text style={styles.error}>{errors.password}</Text>
+              )}
+            </View>
+
+            <TouchableOpacity
+              disabled={!(isValid && dirty)}
+              style={!(isValid && dirty) ? styles.logInDisabled : styles.logIn}
+              onPress={handleSubmit}>
+              <Text style={styles.logInText}> Log In </Text>
+            </TouchableOpacity>
+          </>
+        )}
+      </Formik>
+
+      <ButtonApp
+        label="Sign Up"
+        styleBtn={styles.signUp}
+        styleTxt={styles.signUpText}
+        func={() => signUpPage()}
```

Comment:
```suggestion
        func={signUpPage}
```

Reply:
Though, names for prop and for callback are messy. `func` does not describe anything about when the function is called and what arguments it takes. `signUpPage` sounds like a name for this component, not a function that navigates to that screen. What if we used `onPress` and `goToSignUpScreen`?